source/extensions/filters/network/dubbo_proxy/app_exception.cc

```diff
@@ -0,0 +1,72 @@
+#include "extensions/filters/network/dubbo_proxy/app_exception.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/byte_order.h"
+#include "common/common/hex.h"
+
+#include "extensions/filters/network/dubbo_proxy/message.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+
+AppException::AppException(AppExceptionType type, const std::string& what)
+    : EnvoyException(what), type_(type), response_type_(RpcResponseType::ResponseWithException) {}
+
+AppException::AppException(const AppException& ex) : EnvoyException(ex.what()), type_(ex.type_) {}
+
+AppException::ResponseType AppException::encode(MessageMetadata& metadata,
+                                                DubboProxy::Protocol& protocol,
+                                                Deserializer& deserializer,
+                                                Buffer::Instance& buffer) const {
+  ENVOY_LOG(debug, "err {}", what());
+
+  switch (type_) {
+  case AppExceptionType::ClientTimeout:
```

Comment:
replace AppExceptionType to ResponseStatus?

Obviously, there is not much difference between the AppExceptionType and the ResponseStatus here except for the name.

Reply:
done